walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,45 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
     throw new UnsupportedOperationException("TODO: Implement this method.");
+    
+    //Assigning duration of the meeting 
+    long meetingduration = request.getDuration();
+
+
+    //Saving collection of attendees 
+    Collection<String> meetingattendees = events.getAttendees();
+
+    //Finding all incompatible time ranges
+    ArrayList<TimeRange> incompatime-ranges = FindIncompatibleRanges(events,meetingattendees)
+
+     
+    //Sorting the range by chronological order 
+    Collection.sort(incompatable-ranges,TimeRange.ORDER_BY_START);
+
+
+
+
+
+  }
+
+  private ArrayList<TimeRange> FindIncompatibleRanges(Collection<Event> events, Collection<String> meetingattendees){
+      //Returns a list of times which a meeting cannot occur
+      ArrayList<TimeRange> incompatable-ranges = new Array<TimeRange>();
+      for(Event event : events){
+          TimeRange event-range = event.getWhen();
+          Collection<String> event-attendees = event.getAttendees();
+          if (areMeetingAttendeesInEvent(meetingattendees,event-attendees)){
```

Comment:
Maybe I'm being dense, but I don't see the `areMeetingAttendeesInEvent` function defined anywhere.